### PR TITLE
🚧Create sonar scan status check workflow

### PR DESCRIPTION
**Pull request by kselena**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: .github/workflows/sonar-scan-pr-status-check.yml
Comment (KinsonDigitalAdmin) [inline .github/workflows/sonar-scan-pr-status-check.yml:1]: Please add single empty line to end of the workflow.

> **Note** I figured out what the issue was with the other PR.  It looked like there was an empty line but there was a tab whitespace character on the line.  So the line is not empty.  The extra single line has to be an **EMPTY** line with no whitespace characters such as line feed, carriage return, tab, or space.

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

File: .github/workflows/sonar-scan-pr-status-check.yml
Comment (kselena) [reply .github/workflows/sonar-scan-pr-status-check.yml:1]: There is already an empty line at the end of the workflow...

image

Comment (KinsonDigitalAdmin) [reply .github/workflows/sonar-scan-pr-status-check.yml:1]: Yes but I think there is some content in it.  That content would be empty whitespace.  Did you select the entire empty line to see if anything is selected?  If so there there are spaces and/or tabs in it.  If so, delete the whitespace and try again.

File: .github/workflows/sonar-scan-pr-status-check.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          ./.sonar/scanner/dotnet-sonarscanner begin /k:"KinsonDigital_Carbonate" /o:"kinsondigital" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io"
+
+          dotnet clean "${{ github.workspace }}/Carbonate/Carbonate.csproj";
+          dotnet build "${{ github.workspace }}/Carbonate/Carbonate.csproj" -c Debug;
+
+          ./.sonar/scanner/dotnet-sonarscanner end /d:sonar.login="${{ secrets.SONAR_TOKEN }}"
+
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/sonar-scan-pr-status-check.yml:71]: Please remove the empty whitespace on the single line at the bottom of the file.  You need to remove any whitespace but leave the single empty line behind.

```suggestion

          dotnet clean "${{ github.workspace }}/Carbonate/Carbonate.csproj";
          dotnet build "${{ github.workspace }}/Carbonate/Carbonate.csproj" -c Debug;

          ./.sonar/scanner/dotnet-sonarscanner end /d:sonar.login="${{ secrets.SONAR_TOKEN }}"

```

**VSCode Sample:**
image

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]
